prov/gni/src/gnix_av.c

```diff
@@ -155,40 +157,51 @@ static int table_insert(struct gnix_fid_av *av_priv, const void *addr,
 			size_t count, fi_addr_t *fi_addr, uint64_t flags,
 			void *context)
 {
-	struct gnix_ep_name *temp = NULL;
+	struct gnix_ep_name ep_name;
 	int ret = count;
-	size_t index;
-	size_t i;
+	size_t index, i;
 	int *entry_err = context;
+	void (*get_ep_name)(const char *, int, struct gnix_ep_name *);
 
 	if (gnix_check_capacity(av_priv, count)) {
 		return -FI_ENOMEM;
 	}
 
 	assert(av_priv->table);
+
+	/* Use a function pointer to resolve the address */
+	if (av_priv->domain->addr_format == FI_ADDR_STR) {
+		get_ep_name = _gnix_resolve_str_ep_name;
+	} else {
+		get_ep_name = _gnix_resolve_gni_ep_name;
+	}
```

Comment:
Okay, I'm just thinking that if there are many AV insertions it would be much faster to check for FI_ADDR_STR outside of the loop rather than checking for FI_ADDR_STR each iteration of the loop. But maybe the compiler will unswitch for us or the machines are smart about their branch prediction and the pipeline will only be stalled on the first iteration of the loop. Should I do some small experiments to see what performs better or just remove the function pointer?